lib/library.cpp

```diff
@@ -655,9 +655,12 @@ Library::Error Library::loadFunction(const tinyxml2::XMLElement * const node, co
             func.isconst = true; // a constant function is pure
         } else if (functionnodename == "leak-ignore")
             func.leakignore = true;
-        else if (functionnodename == "use-retval")
+        else if (functionnodename == "use-retval") {
             func.useretval = true;
-        else if (functionnodename == "returnValue") {
+            if (const char *type = functionnode->Attribute("type"))
+                if (std::strcmp(type , "error-code") == 0)
+                    mUseRetvalTypes[name] = Library::RetvalErrorType::ERROR_CODE;
```

Comment:
Instead of adding a map `mUseRetvalTypes` you can modify the `func.useretval` flag.